tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -66,38 +146,478 @@ function Get-AzProfile
   PROCESS
   {
     $Update = $PSBoundParameters.Update
-    # If Update is present, download ProfileMap from Storage blob endpoint
+    # If Update is present, download ProfileMap from online source
     if ($Update.IsPresent)
     {
       return (Get-AzureProfileMap)
     }
 
     # Check the cache
-    $ProfileCache = Get-ProfileCachePath
-    if(Test-Path $ProfileCache)
+    $LatestProfileMapPath = Get-LatestProfileMapPath
+    if(($LatestProfileMapPath -ne $null) -and (Test-Path $LatestProfileMapPath.FullName))
     {
-        $ProfileMap = Get-Content -Raw -Path "$ProfileCache\ProfileMap.json" -ErrorAction SilentlyContinue | ConvertFrom-Json 
-       
-        if ($ProfileMap -ne $null)
-        {
-           return $ProfileMap
-        }
+      $ProfileMap = Get-Content -Raw -Path $LatestProfileMapPath.FullName -ErrorAction SilentlyContinue | ConvertFrom-Json 
+      if ($ProfileMap -ne $null)
+      {
+         return $ProfileMap
+      }
     }
      
     # If cache doesn't exist, Check embedded source
     $defaults = [System.IO.Path]::GetDirectoryName($PSCommandPath)
     $ProfileMap = Get-Content -Raw -Path "$defaults\ProfileMap.json" -ErrorAction SilentlyContinue | ConvertFrom-Json
     if($ProfileMap -eq $null)
     {
-        # Cache & Embedded source empty; Return error and stop
-        throw [System.IO.FileNotFoundException] "$ProfileMap not found."
+      # Cache & Embedded source empty; Return error and stop
+      throw [System.IO.FileNotFoundException] "Profile meta data does not exist. Use 'Get-AzureRmProfile -Update' to download from online source."
     }
 
     return $ProfileMap
   }
 }
 
+# Lists the profiles available for install from gallery
+function Get-ProfilesAvailable
+{
+  param([parameter(Mandatory = $true)] [PSCustomObject] $ProfileMap)
+  $ProfileList = ""
+  foreach ($Profile in $ProfileMap) 
+  {
+    foreach ($Module in ($Profile | get-member -MemberType NoteProperty).Name)
```

Comment:
This is just to simplify & refactor  the "Get-AzureRmProfile" function to make it easier to mock and read.